service/header/core_listener.go

```diff
@@ -56,14 +51,26 @@ func (cl *CoreListener) Start(ctx context.Context) error {
 func (cl *CoreListener) Stop(ctx context.Context) error {
 	cl.cancel()
 	cl.cancel = nil
-	return cl.fetcher.UnsubscribeNewBlockEvent(ctx)
+
+	err := cl.fetcher.UnsubscribeNewBlockEvent(ctx)
+	if err != nil {
+		log.Errorw("core-listener: failed to unsubscribe from new block events", "err", err)
+	}
+	return nil
 }
 
 // listen kicks off a loop, listening for new block events from Core,
 // generating ExtendedHeaders and broadcasting them to the header-sub
 // gossipsub network.
-func (cl *CoreListener) listen(ctx context.Context, sub <-chan *types.Block) {
+func (cl *CoreListener) listen(ctx context.Context) {
 	defer log.Info("core-listener: listening stopped")
+
+	sub, err := cl.fetcher.SubscribeNewBlockEvent(ctx)
+	if err != nil {
+		log.Errorw("core-listener: failed to subscribe to new block events", "err", err)
+		return
+	}
+
```

Comment:
why do we need to take it from the Start? I believe it is ok if the Start fails when a subscription cannot be created. The failed Start prevents the node to be started and the bridge node without the working subscription is useless, thus no need to be started.

Reply:
relic from previous design where sub only started after synced. Re-adding to start.

https://github.com/celestiaorg/celestia-node/pull/507/commits/dfbfdf68e8d84bc68a7502704a21bb28c654d64d

resolved?